### PR TITLE
DOC: read_csv usecols element order is ignored

### LINKED ISSUE #18673: read_csv() usecols behavior unclear
#### Code Sample, a copy-pastable example if possible

```python
                     names= ['Date', 'CreamSupply', 'RawMilk', 'Protein', 'FatProtein', 'MilkFilterInlet' #...... 
],
usecols=
 [0,132,52,19,18,133,71,70,158,92,93,94,156,157,134,135,89,73,72,27,25,24,54,53,75,74,136]

```
#### Problem description

One would think that usecols would allow a user to select the columns (and due to the fact that it's a list), the order of the columns. Either this feature disrespects the order (bug) or it is intended and the documentation should reflect that.

Latest version.

### PR DESCRIPTION
**Pull request by ignamv**
- [x] closes #18673
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -101,7 +101,8 @@
     be positional (i.e. integer indices into the document columns) or strings
     that correspond to column names provided either by the user in `names` or
     inferred from the document header row(s). For example, a valid array-like
-    `usecols` parameter would be [0, 1, 2] or ['foo', 'bar', 'baz'].
+    `usecols` parameter would be [0, 1, 2] or ['foo', 'bar', 'baz']. Element
```
Comment (jreback) [inline pandas/io/parsers.py]: can you also add this note to io.rst

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/parsers.py (hunk 1 above)
Comment (ignamv) [reply pandas/io/parsers.py]: Done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ignamv

**Comment by ignamv** [comment]
No problem.

**Pull request merged**